lib/group_repository.py

```diff
@@ -100,24 +101,29 @@ def _add_hosts_to_group(group_id: str, host_id_list: List[str]):
 
 
 def add_hosts_to_group(group_id: str, host_id_list: List[str], event_producer: EventProducer):
+    custom_staleness = None
     with session_guard(db.session):
         _add_hosts_to_group(group_id, host_id_list)
+        custom_staleness = get_staleness_obj(get_current_identity())
```

Comment:
Does the `custom_staleness` assignment need to happen inside the `session_guard`? Is there a scenario where we want `custom_staleness` to be `None`?
```suggestion
    custom_staleness = get_staleness_obj(get_current_identity())
    with session_guard(db.session):
        _add_hosts_to_group(group_id, host_id_list)
```

Reply:
I don't think it is a hard need. But I would guarantee that we are inside a session to perform this DB interaction!

There is no scenario where `custom_staleness` is `None`